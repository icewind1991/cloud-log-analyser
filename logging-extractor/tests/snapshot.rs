use logging_extractor::{extract_dir, LoggingStatement};
use test_case::test_case;

#[test_case("DefaultShareProvider")]
fn snapshot_test(name: &str) {
    let root = format!("test-data/{name}");
    let mut output = Vec::<u8>::with_capacity(1024 * 1024);
    extract_dir(&root, &mut output).unwrap();
    let output: Vec<LoggingStatement> = serde_json::from_slice(&output).unwrap();

    insta::assert_json_snapshot!(output)
}
