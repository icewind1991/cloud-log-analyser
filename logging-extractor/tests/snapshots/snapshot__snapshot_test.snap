---
source: tests/snapshot.rs
expression: output
---
[
  {
    "level": "error",
    "path": "/DefaultShareProvider.php",
    "line": 1171,
    "message_parts": []
  },
  {
    "level": "debug",
    "path": "/DefaultShareProvider.php",
    "line": 1392,
    "message_parts": [
      "Share notification not sent to ",
      " because user could not be found."
    ]
  },
  {
    "level": "debug",
    "path": "/DefaultShareProvider.php",
    "line": 1401,
    "message_parts": [
      "Share notification not sent to ",
      " because email address is not set."
    ]
  },
  {
    "level": "debug",
    "path": "/DefaultShareProvider.php",
    "line": 1416,
    "message_parts": [
      "Sent share notification to ",
      " for share with ID ",
      "."
    ]
  },
  {
    "level": "error",
    "path": "/DefaultShareProvider.php",
    "line": 1420,
    "message_parts": [
      "Share notification mail could not be sent."
    ]
  },
  {
    "level": "error",
    "path": "/DefaultShareProvider.php",
    "line": 1507,
    "message_parts": [
      "Share notification mail could not be sent to: ",
      ", "
    ]
  }
]
